### PR TITLE
Add io.anytype.anytype

### PR DESCRIPTION
**Pull request by mhymny**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
Anytype is a local-first, E2E encrypted software designed as a secure place to create and store digital assets.

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. https://github.com/anyproto/anytype-ts/issues/74#issuecomment-2354593310

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.anytype.anytype.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      - type: script
+        commands:
+          - export TMPDIR="${XDG_RUNTIME_DIR}/app/${FLATPAK_ID}"
+          - exec zypak-wrapper /app/Anytype/anytype "$@"
+        dest-filename: io.anytype.anytype.sh
+rename-desktop-file: anytype.desktop
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+separate-locales: false
```
Comment (hfiguiere) [inline io.anytype.anytype.yml:45]: these lines 41-45 should be above `command`.

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: no arrch64?

File: io.anytype.anytype.svg
Comment (hfiguiere) [inline io.anytype.anytype.svg:1]: the .deb has no icon? It's a bug.

File: io.anytype.anytype.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  - --env=ELECTRON_OZONE_PLATFORM_HINT=wayland
+modules:
+  - shared-modules/libsecret/libsecret.json
+  - name: anytype
+    buildsystem: simple
+    build-commands:
+      - ar -o -x anytype_amd64.deb
+      - tar --extract --file data.tar.xz
+      - cp --preserve --recursive opt/* usr/* /app
+      - desktop-file-edit --set-key Exec --set-value "/app/bin/anytype %U" /app/share/applications/anytype.desktop
```
Comment (hfiguiere) [inline io.anytype.anytype.yml:20]:
```suggestion
      - desktop-file-edit --set-key Exec --set-value "anytype %U" /app/share/applications/anytype.desktop
```

File: io.anytype.anytype.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      - desktop-file-edit --set-key Icon --set-value "io.anytype.anytype" /app/share/applications/anytype.desktop
```
Comment (hfiguiere) [inline io.anytype.anytype.yml:21]: `rename-icon` use like `rename-desktop-file` should do the trick.

File: flathub.json (hunk 2 above)
Comment (mhymny) [reply flathub.json:2]: Not that I am aware of. There is a mac specific arm64 release, however I have nothing to test this against. I will try to build from source in the future.

File: io.anytype.anytype.svg
Comment (mhymny) [reply io.anytype.anytype.svg:1]: There are icons, just not svg.

File: io.anytype.anytype.yml (hunk 4 above)
Comment (mhymny) [reply io.anytype.anytype.yml:21]: Im sorry, I don't understand what you mean, can you explain?

File: io.anytype.anytype.svg
Comment (hfiguiere) [reply io.anytype.anytype.svg:1]: then remove this.

File: io.anytype.anytype.yml (hunk 4 above)
Comment (hfiguiere) [reply io.anytype.anytype.yml:21]: `rename-icon` that you should use to use the icons provided by the binary pacakge WILL update the .desktop file, making this unecessary. Works like `rename-desktop-file`

Comment (mhymny) [reply io.anytype.anytype.yml]: Okay, removed the svg and renamed the icons using `rename-icon`.

**Comment by hfiguiere** [comment]
bot, build io.anytype.anytype

**Comment by mhymny** [comment]
bot, build io.anytype.anytype

**Comment by mhymny** [comment]
bot, build io.anytype.anytype

File: io.anytype.anytype.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>io.anytype.anytype</id>
+
+  <name>Anytype</name>
+  <summary>The everything app for those who celebrate trust and autonomy</summary>
+
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>LicenseRef-proprietary=https://raw.githubusercontent.com/anyproto/anytype-ts/main/LICENSE.md</project_license>
```
Comment (bbhtt) [inline io.anytype.anytype.metainfo.xml:9]: Does this allow redistribution ? If not you should use extra-data source unless upstream is fine with using the original source.

Comment (mhymny) [reply io.anytype.anytype.metainfo.xml:9]: It does allow distribution in a non-commercial way:

> Any Association grants you (“Licensee”) a license to use, modify, and 
redistribute the Software, but only 
(a) for Non-Commercial Use, or (b) for Commercial Use in Allowed Networks.

And the upstream is fine with me [publishing](https://github.com/anyproto/anytype-ts/issues/74)
> we're fine with it. Let's see if the users will be satisfied with this solution or if they want it to be distributed from our account :)

**Pull request closed without merging**